ts/windows/attachments.ts

```diff
@@ -230,14 +230,26 @@ async function writeWithAttributes(
 export const saveAttachmentToDisk = async ({
   data,
   name,
+  baseDir,
 }: {
   data: Uint8Array;
   name: string;
+  /**
+   * Base directory for saving the attachment.
+   * If omitted, a dialog will be opened to let the user choose a directory
+   */
+  baseDir?: string;
 }): Promise<null | { fullPath: string; name: string }> => {
-  const { canceled, filePath } = await showSaveDialog(name);
+  let filePath;
 
-  if (canceled || !filePath) {
-    return null;
+  if (!baseDir) {
+    const { canceled, filePath: dialogFilePath } = await showSaveDialog(name);
+    if (canceled || !dialogFilePath) {
+      return null;
```

Comment:
We probably want to throw if the user didn't cancel but we got a null path back.

Reply:
I added the throw, please check if you like the error message

Reply:
That seems reasonable.